src/java/org/apache/cassandra/service/accord/txn/TxnUpdate.java

```diff
@@ -339,7 +339,7 @@ private static <T> List<T> deserialize(ByteBuffer bytes, IVersionedSerializer<T>
 
         try (DataInputBuffer in = new DataInputBuffer(bytes, true))
         {
-            int version = in.readUnsignedVInt32();
+            Version version = Version.fromVersion(in.readUnsignedVInt32());
```

Comment:
punted on fixing in this patch as its kinda annoying, so can try to fix after the other open issues are addressed?